viewing_party/main.py

```diff
@@ -0,0 +1,172 @@
+
+###################### WAVE 1 PASSED #########################
+### Wave 1.1 (4 tests - PASSED) ###
+def create_movie(movie_title, genre, rating):
+    
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating,
+    }       
+    
+    if new_movie["title"] == None:
+        new_movie = None
+    elif new_movie["genre"] == None:
+        new_movie = None
+    elif new_movie["rating"] == None:
+        new_movie = None
+    return new_movie
+
+### Wave 1.2 (1 test PASSED) ###
+def add_to_watched(user_data, movie):    
+    # user_data is a dictionary key"watched": value lists title,genre,rating
+    # updated_data is a list of dictionaries
+    user_data["watched"].append(movie)
+    return user_data
+
+### Wave 1.3 (1 test PASSED) ###
+def add_to_watchlist(user_data, movie): 
+    user_data["watchlist"].append(movie)
+    return user_data
+
+### Wave 1.4 (3 tests PASSED) ###
+def watch_movie(user_data, title):
+    for i in user_data["watchlist"]:
+        if i["title"] == title:
+            user_data["watchlist"].remove(i)
+            user_data["watched"].append(i)
+    return user_data
+
+
+###################### WAVE 2 PASSED #########################
+### Wave 2.1 (2 tests) PASSED ###
+def get_watched_avg_rating(user_data):
+    watched_count = 0
+    rating_sum = 0.0
+    average = 0.0
+    if user_data["watched"] == []:
+        return average
+    else:
+        for i in user_data["watched"]:
+            watched_count += 1
```

Comment:
Just a note that you can also use `len(user_data["watched"])`, instead of making a count variable.  This still works and is fine, using `len` is just more compact.